src/index.ts

```diff
@@ -591,14 +617,14 @@ exports.isValidSignature = function (v, r, s, homestead, chainId) {
     return false
   }
 
-  r = new BN(r)
-  s = new BN(s)
+  const rBN: BN = new BN(r)
+  const sBN: BN = new BN(s)
```

Comment:
In `generateAddress2` shadowing variables is not a problem. Any reason to make a distinction here?

Reply:
I couldn't find how to shadow a variable with a different type without the typescript compiler complaining. In `generateAddress2` the type stays the same.

Reply:
If the type stays the same then there's no point to have that expression in the first place :)